kernel/kernel-api/src/main/java/io/delta/kernel/internal/TableConfig.java

```diff
@@ -260,11 +284,11 @@ public static Map<String, String> validateProperties(
       String value = kv.getValue();
       if (key.startsWith("delta.")) {
         TableConfig<?> tableConfig = VALID_PROPERTIES.get(key);
-        if (tableConfig != null) {
+        if (tableConfig.editable) {
           tableConfig.validate(engine, value);
           validatedConfigurations.put(tableConfig.getKey(), value);
         } else {
-          throw DeltaErrors.unknownConfigurationException(key);
+          throw DeltaErrors.cannotModifyTableProperty(key);
```

Comment:
lets use the `kv.getKey()`, basically the connector provided case string in error messages, so that it is clear what property has the problem.